x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/XPackSettings.java

```diff
@@ -117,6 +117,11 @@ private XPackSettings() {
     /** Setting for enabling or disabling vectors. Defaults to true. */
     public static final Setting<Boolean> VECTORS_ENABLED = Setting.boolSetting("xpack.vectors.enabled", true, Setting.Property.NodeScope);
 
+    /** Setting for enabling or disabling the voting-only-node functionality. Needs to be enabled on both voting-only nodes and regular
+     *  master-eligible nodes for the voting-only functionality to work correctly. Defaults to true. */
+    public static final Setting<Boolean> VOTING_ONLY_ENABLED = Setting.boolSetting("xpack.voting_only.enabled", true,
```

Comment:
Why is it necessary to have a separate enabling of the feature? Why coudln't it just always be available with xpack (especially since you have the default as enabled)? Why would someone need to disable this feature, when they would just not set any nodes to voting only?

Reply:
Good question. My thinking was the following: We have this setting for all other features as well. When the setting is disabled, the other plugins are not adding anything to the extension points. I thought this could be used as an extra safeguard that would allow completely disabling the functionality provided by this plugin in case something was wrong with it, given that it's always bundled with x-pack. I'm fine not having this setting if you think that that is not necessary. I did not see a strong reason to deviate from the norm here.

Reply:
IMO it contributes to an overload of settings users could mess with, with no real added benefit. I see what you mean about all the other xpack plugins. I wonder if we really need that though. For eg security it makes sense, but why would one ever disable deprecation, or rollup, or ilm? These features do not actively do anything unless used, afaik. For features like that, I think we should just have it always "enabled", but I'm open to hearing others arguments for why we need the ability to disable across all xpack features.

Reply:
For features that provide API endpoints, it makes perhaps more sense (so that these API endpoints don't even become available). For voting_only nodes, the only thing that becomes visible is that the `voting_only` role appears in several stats endpoints, even if there is no voting_only node in the cluster. The node counts, that are part of the cluster stats, for example, expose this info.
I don't hold a particularly strong opinion either way on this, so I've removed the setting in bcd6ec2. I can revert that if others feel differently about this.